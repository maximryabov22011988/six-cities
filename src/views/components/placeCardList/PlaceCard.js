import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cn from 'classnames';

import Label from '../Label';
import Button from '../Button';
import Image from '../Image';
import Rating from '../Rating';
import SvgIcon from '../SvgIcon';

import withActiveItem from '../../hocs/withActiveItem';

const propTypes = {
  className: PropTypes.string,
  offer: PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.object,
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatar_url: PropTypes.string,
      id: PropTypes.number,
      is_pro: PropTypes.bool,
      name: PropTypes.string,
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    is_favorite: PropTypes.bool,
    is_premium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
    max_adults: PropTypes.number,
    preview_image: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string,
  }).isRequired,
  onActiveOfferClick: PropTypes.func.isRequired,
};

function PlaceCard({ className, offer, onActiveOfferClick, onMouseEnter, onMouseLeave }) {
  const {
    id,
    title,
    preview_image: previewImage,
    price,
    type,
    rating,
    is_premium: isPremium,
    is_favorite: isFavorite,
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  return (
    <article
      className={cn(
        'place-card',
        className === 'cities' && `${className}__place-card`,
        className === 'near-places' && `${className}__card`,
      )}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      <Label isShow={isPremium} name="Premium" parentClassName="place-card" />

      <div className={`${className}__image-wrapper place-card__image-wrapper`}>
        <a href="#" onClick={onActiveOfferClick}>
          <Image className="place-card__image" height="200" label="Place image" src={previewImage} width="260" />
        </a>
      </div>

      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>

          <BookmarkButton className="place-card__bookmark-button" isActive={isFavorite}>
            <SvgIcon
              className="place-card__bookmark-icon"
              height="19"
              isShowLabel={false}
              label={isFavorite ? 'In bookmarks' : 'To bookmarks'}
              name="icon-bookmark"
              width="18"
            />
          </BookmarkButton>
        </div>

        <Rating parentClassName="place-card" rating={rating} />

        <h2 className="place-card__name">
          <Link to={`/offer/${id}`}>{title}</Link>
        </h2>

        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
}

PlaceCard.propTypes = propTypes;

export default PlaceCard;
