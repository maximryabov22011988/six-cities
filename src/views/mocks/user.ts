export default {
  avatar_url: '/static/avatar/8.jpg',
  email: 'Oliver.conner@gmail.com',
  id: 1,
  is_pro: false,
  name: 'Oliver.conner',
};
