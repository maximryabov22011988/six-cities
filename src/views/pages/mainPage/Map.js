import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import Leaflet from 'leaflet';

const mapStyles = {
  top: '15px',
  height: '794px'
};

const LEAFLET_LAYER = {
  URL_TEMPLATE:
    'https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png',
  OPTIONS: {
    attribution:
      '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>'
  }
};

// const propTypes = {
//   offers: PropTypes.arrayOf(
//     PropTypes.shape({
//       id: PropTypes.string,
//       title: PropTypes.string,
//       image: PropTypes.string,
//       price: PropTypes.number,
//       type: PropTypes.oneOf(['Apartment', 'Private room']),
//       rating: PropTypes.number,
//       isPremium: PropTypes.bool,
//       isBookmark: PropTypes.bool,
//       location: PropTypes.arrayOf(PropTypes.number)
//     })
//   ).isRequired
// };

class Map extends Component {
  constructor(props) {
    super(props);

    this.leaflet = Leaflet;

    this.icon = this.leaflet.icon({
      iconUrl: 'img/pin.svg',
      iconSize: [27, 39],
      iconAnchor: [13, 39]
    });
  }

  componentDidMount() {
    const { offers } = this.props;

    if (offers.length > 0) {
      this.map = this.leaflet.map(
        'map',
        {
          center: [
            offers[0].city.location.latitude,
            offers[0].city.location.longitude
          ],
          zoom: offers[0].city.location.zoom,
          zoomControl: false,
          layers: new this.leaflet.TileLayer(
            LEAFLET_LAYER.URL_TEMPLATE,
            LEAFLET_LAYER.OPTIONS
          )
        },
        100
      );

      this.markers = this.leaflet.layerGroup().addTo(this.map);
      this.addMarkers(offers, this.markers);
    }
  }

  componentDidUpdate() {
    const { offers } = this.props;

    if (this.map) {
      this.map.setView(
        [offers[0].city.location.latitude, offers[0].city.location.longitude],
        offers[0].city.location.zoom
      );
      this.markers.clearLayers();
      this.addMarkers(offers, this.markers);
    }
  }

  addMarkers(offers, group) {
    offers.map(offer => {
      this.leaflet
        .marker([offer.location.latitude, offer.location.longitude], {
          icon: this.icon
        })
        .addTo(group);
    });
  }

  render() {
    return <div id="map" style={mapStyles} />;
  }
}

// Map.propTypes = propTypes;

export default Map;
