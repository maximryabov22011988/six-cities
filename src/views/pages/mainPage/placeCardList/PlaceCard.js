import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import withActiveItem from '../../../hocs/withActiveItem';
import Button from '../../../components/Button';
import Rating from './placeCard/Rating';

const propTypes = {
  offer: PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.object
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatar_url: PropTypes.string,
      id: PropTypes.number,
      is_pro: PropTypes.bool,
      name: PropTypes.string
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    is_favorite: PropTypes.bool,
    is_premium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number
    }),
    max_adults: PropTypes.number,
    preview_image: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  onTitleClick: PropTypes.func.isRequired,
  onImageClick: PropTypes.func.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired
};

function PlaceCard(props) {
  const {
    className,
    offer,
    onTitleClick,
    onImageClick,
    onMouseEnter,
    onMouseLeave
  } = props;

  const {
    title,
    preview_image: previewImage,
    price,
    type,
    rating,
    is_premium: isPremium,
    is_favorite: isFavorite
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  return (
    <article
      className={classnames('cities__place-card place-card', className)}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {isPremium && (
        <div className="place-card__mark">
          <span>Premium</span>
        </div>
      )}

      <div className="cities__image-wrapper place-card__image-wrapper">
        <a href="#" onClick={onImageClick(offer)}>
          <img
            className="place-card__image"
            src={previewImage}
            width="260"
            height="200"
            alt="Place image"
          />
        </a>
      </div>

      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>

          <BookmarkButton
            className="place-card__bookmark-button"
            isActive={isFavorite}
          >
            <svg className="place-card__bookmark-icon" width="18" height="19">
              <use xlinkHref="#icon-bookmark" />
            </svg>
            <span className="visually-hidden">To bookmarks</span>
          </BookmarkButton>
        </div>

        <Rating rating={rating} />

        <h2 className="place-card__name">
          <a href="#" onClick={onTitleClick}>
            {title}
          </a>
        </h2>

        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
}

PlaceCard.propTypes = propTypes;

export default PlaceCard;
