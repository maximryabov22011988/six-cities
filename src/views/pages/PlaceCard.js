import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import Header from '../components/Header';
import Logo from '../components/Logo';
import UserInfo from '../components/UserInfo';
import Label from '../components/Label';
import Rating from '../components/Rating';
import Button from '../components/Button';
import SvgIcon from '../components/SvgIcon';
import Avatar from '../components/Avatar';
import Map from '../components/Map';

import Reviews from './placeCard/Reviews';
import NearPlaces from './placeCard/NearPlaces';
import AddReviewsForm from './placeCard/AddReviewsForm';

import withActiveItem from '../hocs/withActiveItem';

import { sendReview } from '../../state/reviews/operations';
import { getCurrentOffer, getNearOffers } from '../../state/offers/selectors';

const MAX_IMAGES = 6;

const propTypes = {
  addOfferToFavorities: PropTypes.func.isRequired,
  isAuthUser: PropTypes.bool.isRequired,
  nearOffers: PropTypes.arrayOf(PropTypes.object),
  offer: PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.object,
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatar_url: PropTypes.string,
      id: PropTypes.number,
      is_pro: PropTypes.bool,
      name: PropTypes.string,
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    is_favorite: PropTypes.bool,
    is_premium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
    max_adults: PropTypes.number,
    preview_image: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string,
  }).isRequired,
  removeOfferFromFavorities: PropTypes.func.isRequired,
  sendReview: PropTypes.func.isRequired,
  user: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
    name: PropTypes.string,
    avatar_url: PropTypes.string,
    is_pro: PropTypes.bool,
  }).isRequired,
};

function PlaceCard({
  isAuthUser,
  user: { avatar_url: userAvatarUrl, email },
  nearOffers,
  offer,
  sendReview,
  addOfferToFavorities,
  removeOfferFromFavorities,
}) {
  const {
    id,
    images,
    title,
    price,
    type,
    bedrooms,
    host: { avatar_url: avatarUrl, is_pro: isPro, name },
    is_favorite: isFavorite,
    is_premium: isPremium,
    max_adults: maxAdults,
    rating,
    goods,
    description,
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  const handleAddToFavorities = (hotelId) => () => addOfferToFavorities(hotelId);
  const handleRemoveFromFavorities = (hotelId) => () => removeOfferFromFavorities(hotelId);

  return (
    <div className="page">
      <Header
        logo={<Logo position="header" />}
        userInfo={<UserInfo avatarUrl={userAvatarUrl} email={email} isAuth={isAuthUser} />}
      />

      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {images.map((src, i) => {
                if (i < MAX_IMAGES) {
                  return (
                    <div className="property__image-wrapper" key={i}>
                      <img alt="Hotel interior" className="property__image" src={src} />
                    </div>
                  );
                }
                return null;
              })}
            </div>
          </div>

          <div className="property__container container">
            <div className="property__wrapper">
              <Label isShow={isPremium} name="Premium" parentClassName="property" />

              <div className="property__name-wrapper">
                <h1 className="property__name">{title}</h1>
                <BookmarkButton
                  className="property__bookmark-button"
                  isActive={isFavorite}
                  onClick={isFavorite ? handleRemoveFromFavorities(id) : handleAddToFavorities(id)}
                >
                  <SvgIcon
                    className="property__bookmark-icon"
                    height="33"
                    isShowLabel={false}
                    label={isFavorite ? 'In bookmarks' : 'To bookmarks'}
                    name="icon-bookmark"
                    width="31"
                  />
                </BookmarkButton>
              </div>

              <Rating parentClassName="property" rating={rating} showValue />

              <ul className="property__features">
                <li className="property__feature property__feature--entire">{type}</li>
                <li className="property__feature property__feature--bedrooms">{bedrooms} Bedrooms</li>
                <li className="property__feature property__feature--adults">Max {maxAdults} adults</li>
              </ul>

              <div className="property__price">
                <b className="property__price-value">&euro;{price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>

              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {goods.map((good, i) => (
                    <li className="property__good-item" key={i}>
                      {good}
                    </li>
                  ))}
                </ul>
              </div>

              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>

                <div className="property__host-user user">
                  <Avatar
                    alt="Host avatar"
                    height="74"
                    isPro={isPro}
                    name={name}
                    parentClassName="property"
                    showStatus
                    src={avatarUrl}
                    width="74"
                  />
                </div>

                <div className="property__description">
                  <p className="property__text">{description}</p>
                </div>
              </div>

              <section className="property__reviews reviews">
                <Reviews hotelId={id} />
                {isAuthUser && <AddReviewsForm hotelId={id} onSendReview={sendReview} />}
              </section>
            </div>
          </div>
          <Map className="property__map" currentOffer={offer} fixed offers={nearOffers} />
        </section>

        <NearPlaces
          offers={nearOffers}
          onAddToFavorities={handleAddToFavorities}
          onRemoveFromFavorities={handleRemoveFromFavorities}
        />
      </main>
    </div>
  );
}

PlaceCard.propTypes = propTypes;

const mapStateToProps = (state, ownProps) => ({
  offer: getCurrentOffer(state, ownProps),
  nearOffers: getNearOffers(state, ownProps),
});

const mapDispatchToProps = {
  sendReview,
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(PlaceCard)
);
