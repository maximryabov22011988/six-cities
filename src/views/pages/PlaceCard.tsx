import * as React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import Header from '../components/Header';
import Logo from '../components/Logo';
import UserInfo from '../components/UserInfo';
import Label from '../components/Label';
import Rating from '../components/Rating';
import Button from '../components/Button';
import SvgIcon from '../components/SvgIcon';
import Avatar from '../components/Avatar';
import Map from '../components/Map';

import Reviews from './placeCard/Reviews';
import NearPlaces from './placeCard/NearPlaces';
import AddReviewsForm from './placeCard/AddReviewsForm';

import withActiveItem from '../hocs/withActiveItem';

import { sendReview } from '../../state/reviews/operations';
import { getCurrentOffer, getNearOffers } from '../../state/offers/selectors';

import { User, Offer, onAddToFavorities, onRemoveFromFavorities } from '../interfaces';

const MAX_IMAGES = 6;

interface Props {
  addOfferToFavorities: onAddToFavorities,
  isAuthUser: boolean,
  nearOffers: Array<Offer>,
  offer: Offer,
  removeOfferFromFavorities: onRemoveFromFavorities,
  sendReview: () => void,
  user:User,
}

function PlaceCard({
  isAuthUser,
  user: { avatar_url: userAvatarUrl, email },
  nearOffers,
  offer,
  sendReview,
  addOfferToFavorities,
  removeOfferFromFavorities,
}: Props) {
  const {
    id,
    images,
    title,
    price,
    type,
    bedrooms,
    host: { avatar_url: avatarUrl, is_pro: isPro, name },
    is_favorite: isFavorite,
    is_premium: isPremium,
    max_adults: maxAdults,
    rating,
    goods,
    description,
  } = offer;

  const BookmarkButton = withActiveItem(Button);

  const handleAddToFavorities = (hotelId) => () => addOfferToFavorities(hotelId);
  const handleRemoveFromFavorities = (hotelId) => () => removeOfferFromFavorities(hotelId);

  return (
    <div className="page">
      <Header
        logo={<Logo position="header" />}
        userInfo={<UserInfo avatarUrl={userAvatarUrl} email={email} isAuth={isAuthUser} />}
      />

      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {images.map((src, i) => {
                if (i < MAX_IMAGES) {
                  return (
                    <div className="property__image-wrapper" key={i}>
                      <img alt="Hotel interior" className="property__image" src={src} />
                    </div>
                  );
                }
                return null;
              })}
            </div>
          </div>

          <div className="property__container container">
            <div className="property__wrapper">
              <Label isShow={isPremium} name="Premium" parentClassName="property" />

              <div className="property__name-wrapper">
                <h1 className="property__name">{title}</h1>
                <BookmarkButton
                  className="property__bookmark-button"
                  isActive={isFavorite}
                  onClick={isFavorite ? handleRemoveFromFavorities(id) : handleAddToFavorities(id)}
                >
                  <SvgIcon
                    className="property__bookmark-icon"
                    height="33"
                    isShowLabel={false}
                    label={isFavorite ? 'In bookmarks' : 'To bookmarks'}
                    name="icon-bookmark"
                    width="31"
                  />
                </BookmarkButton>
              </div>

              <Rating isShowValue parentClassName="property" rating={rating} />

              <ul className="property__features">
                <li className="property__feature property__feature--entire">{type}</li>
                <li className="property__feature property__feature--bedrooms">{bedrooms} Bedrooms</li>
                <li className="property__feature property__feature--adults">Max {maxAdults} adults</li>
              </ul>

              <div className="property__price">
                <b className="property__price-value">&euro;{price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>

              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {goods.map((good, i) => (
                    <li className="property__good-item" key={i}>
                      {good}
                    </li>
                  ))}
                </ul>
              </div>

              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>

                <div className="property__host-user user">
                  <Avatar
                    alt="Host avatar"
                    height="74"
                    isPro={isPro}
                    name={name}
                    parentClassName="property"
                    showStatus
                    src={avatarUrl}
                    width="74"
                  />
                </div>

                <div className="property__description">
                  <p className="property__text">{description}</p>
                </div>
              </div>

              <section className="property__reviews reviews">
                <Reviews hotelId={id} />
                {isAuthUser && <AddReviewsForm hotelId={id} onSendReview={sendReview} />}
              </section>
            </div>
          </div>
          <Map className="property__map" currentOffer={offer} fixed offers={nearOffers} />
        </section>

        <NearPlaces
          offers={nearOffers}
          onAddToFavorities={handleAddToFavorities}
          onRemoveFromFavorities={handleRemoveFromFavorities}
        />
      </main>
    </div>
  );
}

const mapStateToProps = (state, ownProps) => ({
  offer: getCurrentOffer(state, ownProps),
  nearOffers: getNearOffers(state, ownProps),
});

const mapDispatchToProps = {
  sendReview,
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(PlaceCard)
);
