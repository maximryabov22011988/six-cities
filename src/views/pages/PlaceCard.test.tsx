import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";
import axiosMock from 'axios';

import { PlaceCard } from './PlaceCard';
import Map from '../components/Map';

import user from '../mocks/user';
import offer from '../mocks/offer';

import reviewsReducer from '../../state/reviews';
import nameSpace from '../../state/name-spaces';

const reducer = combineReducers({
  [nameSpace.REVIEWS]: reviewsReducer
});
const store = createStore(reducer, applyMiddleware(thunk.withExtraArgument(axiosMock)));

it('PlaceCard renders correctly with mandatory props', () => {
  Map.prototype.componentDidMount = jest.fn();

  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <PlaceCard
            addOfferToFavorities={jest.fn()}
            isAuthUser={true}
            nearOffers={[offer]}
            offer={offer}
            removeOfferFromFavorities={jest.fn()}
            sendReview={jest.fn()}
            user={user}
          />
        </BrowserRouter>
      </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
