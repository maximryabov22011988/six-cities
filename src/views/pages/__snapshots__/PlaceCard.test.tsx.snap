// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard renders correctly with mandatory props 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper false"
                >
                  <img
                    alt="User avatar"
                    className="reviews__avatar user__avatar"
                    src="https://es31-server.appspot.com/six-cities/static/avatar/8.jpg"
                  />
                </div>
                <span
                  className="header__user-name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Hotel interior"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/18.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Hotel interior"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/18.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Hotel interior"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/1.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Hotel interior"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/8.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Hotel interior"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/12.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Hotel interior"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/11.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Canal View Prinsengracht
            </h1>
            <button
              className="property__bookmark-button button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "54.00%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              2.7
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              house
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              4
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              5
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              161
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__good-item"
              >
                Breakfast
              </li>
              <li
                className="property__good-item"
              >
                Laptop friendly workspace
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
              >
                <img
                  alt="Host avatar"
                  className="reviews__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
            <ul
              className="reviews__list"
            />
            <form
              className="reviews__form form"
              method="post"
              onSubmit={[Function]}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="5"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="4"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="3"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="2"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  id="1-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="1"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-stars"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                onChange={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
                value=""
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50
                     characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <div
        className="property__map"
        id="map"
        style={
          Object {
            "height": "580px",
            "marginLeft": "auto",
            "marginRight": "auto",
            "top": 0,
            "width": "1144px",
            "zIndex": 100,
          }
        }
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="places__list near-places__list"
        >
          <article
            className="place-card near-places__card"
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "background": "#e1e1e1",
                      "borderRadius": "4px",
                      "display": "flex",
                      "height": "200px",
                      "justifyContent": "center",
                      "width": "260px",
                    }
                  }
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height={0}
                    onError={[Function]}
                    onLoad={[Function]}
                    src="https://es31-server.appspot.com/six-cities/static/hotel/17.jpg"
                    width={0}
                  />
                  <span>
                    Loading ...
                  </span>
                </div>
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    161
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "54.00%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/1"
                  onClick={[Function]}
                >
                  Canal View Prinsengracht
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                house
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
